scipy/stats/tests/test_multivariate.py

```diff
@@ -349,17 +459,56 @@ def test_exception_non_psd_cov(self):
         cov = [[1, 0], [0, -1]]
         assert_raises(ValueError, _PSD, cov)
 
+    def _assert_methods_raise_exception(self, exc, x, **args):
+        assert_raises(exc, multivariate_normal, **args)
+        assert_raises(exc, multivariate_normal.pdf, x, **args)
+        assert_raises(exc, multivariate_normal.logpdf, x, **args)
+        assert_raises(exc, multivariate_normal.cdf, x, **args)
+        assert_raises(exc, multivariate_normal.logcdf, x, **args)
+
     def test_exception_singular_cov(self):
         np.random.seed(1234)
         x = np.random.randn(5)
         mean = np.random.randn(5)
         cov = np.ones((5, 5))
-        e = np.linalg.LinAlgError
-        assert_raises(e, multivariate_normal, mean, cov)
-        assert_raises(e, multivariate_normal.pdf, x, mean, cov)
-        assert_raises(e, multivariate_normal.logpdf, x, mean, cov)
-        assert_raises(e, multivariate_normal.cdf, x, mean, cov)
-        assert_raises(e, multivariate_normal.logcdf, x, mean, cov)
+        inverse_cov = 1 / cov
+        self._assert_methods_raise_exception(
+            np.linalg.LinAlgError,
+            x, mean=mean, cov=cov
+        )
+        # Check when mean and inverse covariance are specified.
+        self._assert_methods_raise_exception(
+            np.linalg.LinAlgError,
+            x, mean=mean, cov=cov, inverse_cov=inverse_cov
+        )
+
+    def test_exception_singular_inverse_cov(self):
+        np.random.seed(1234)
+        x = np.random.randn(5)
+        mean = np.random.randn(5)
+        cov = np.ones((5, 5))
+        inv_cov = 1 / cov
+        # Check when mean and inverse covariance are specified.
+        self._assert_methods_raise_exception(
+            np.linalg.LinAlgError,
+            x, mean=mean, cov=None, inverse_cov=inv_cov
+        )
```

Comment:
I don't think we'll need these anymore when there's no longer an `inverse_cov` input?